### PR TITLE
Various strategies for cut, and profiling timers

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: api/shape/startTimer.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Shape from './Shape.js';
+
+export const starts = new Map();
+
+export const startTimer = (name) => (shape) => {
```
Comment (BarbourSmith) [inline api/shape/startTimer.js:5]: This is a nice feature to have generally

File: geometry/tagged/difference.js [hunk 2]
```diff
@@ -1,17 +1,56 @@
...
-import { getGraphs } from './getGraphs.js';
 import { difference as graphDifference } from '../graph/difference.js';
 import { rewrite } from './visit.js';
 import { toConcreteGeometry } from './toConcreteGeometry.js';
 
-export const difference = (geometry, ...geometries) => {
-  geometries = geometries.map(toConcreteGeometry);
+export const difference = (geometry, options = {}, ...geometries) => {
+  if (
+    !['incremental', 'single', 'single_recursive', 'basic', undefined].includes(
```
Comment (BarbourSmith) [inline geometry/tagged/difference.js:15]: So all of the modes are exposed now in the api which is nice for testing. Long term do you think all of them will be exposed or if there is a clear winner will just one be exposed in the end?

**Review by BarbourSmith — APPROVED** [review]
👍 👍 😀 

I'm going to have to update my local copy and play with these

File: api/shape/startTimer.js (hunk 1 above)
Comment (pentacular) [reply api/shape/startTimer.js:5]: Yeah, I thought I had this already, but I couldn't find it. :)

**Pull request merged**